server/matchmaker/matchmaker_queue.py

```diff
@@ -102,6 +102,8 @@ async def queue_pop_timer(self) -> None:
 
             number_of_matches = len(self._matches)
             metrics.matches.labels(self.name).set(number_of_matches)
+            number_of_searches_in_queue = len(self.queue)
```

Comment:
...and is `number_of_searches_in_queue` then equal to `number_of_unmatched_searches`?
I cannot tell whether `number_of_searches_in_queue` is meant to report the number of searches before matching was done or after. As it is now I believe it is reporting "after", which we already report.

I think we are currently reporting the number of matches made (during the current queue pop),
as well as the number of searches remaining in the queue after the queue pop.
The number of searches in the queue before matching would then (currently) be `2*number_of_matches + number_of_unmatched_searches`, although that won't hold anymore as soon as the matchmaker handles non-2v2 games.

What you could do instead is update the `searches_in_queue` metric in `MatchmakerQueue.search`, to have a value that is reported not only during queue pops, but also in between?
I'm thinking of these lines:
https://github.com/FAForever/server/blob/a024a2fe8c8b7b905a5bdd41795f8134ee01e677/server/matchmaker/matchmaker_queue.py#L134-L135
https://github.com/FAForever/server/blob/a024a2fe8c8b7b905a5bdd41795f8134ee01e677/server/matchmaker/matchmaker_queue.py#L144-L145

Reply:
Would we then expect the sum over all matchmaker name labels to be the what `ladder_service_searches_dict_length` is currently reporting? Equivalently, is every `ladder_service` search being registered with a `MatchmakerQueue` or can they ever get out of sync?

It might still be good to report `ladder_service_searches_dict_length` separately as you are currently suggesting, in case there will be some bug causing them to get out of sync even if we don't expect them to.